ocaml/runtime/domain.c

```diff
@@ -686,22 +647,12 @@ static void domain_create(uintnat initial_minor_heap_wsize,
   caml_state = domain_state;
 
   domain_state->young_limit = 0;
-  /* Synchronized with [caml_interrupt_all_signal_safe], so that the
-     initializing write of young_limit happens before any
-     interrupt. */
-  atomic_store_explicit(&s->interrupt_word, &domain_state->young_limit,
-                        memory_order_release);
-
-  /* Tell memprof system about the new domain before either (a) new
-   * domain can allocate anything or (b) parent domain can go away. */
-  caml_memprof_new_domain(parent, domain_state);
-  if (!domain_state->memprof) {
-    goto init_memprof_failure;
-  }
 
->>>>>>> 5.2.0
+  // XXX mshinwell: check if this unique_id line is needed and whether
+  // these two lines should be before caml_memprof_new_domain (as we had on
+  // flambda-backend) or after (as on 5.2, although 5.2 lacks the unique_id
+  // line).
```

Comment:
The `id` needs to be initialised before `caml_memprof_new_domain`. This code should look like upstream trunk and like current flambda-backend, both of which include the new statmemprof code. (Upstream 5.2.0 lacks this)